### DISCUSSION TITLE
node test runner: Filtering individual tests in a heirarchy

### DISCUSSION THREAD

**Discussion by rotty3000** [nodejs/help#4815] [Q&A]
**tldr;**
Is there a way to filter to a specific test in a test hierarchy? `--test-name-pattern=` doesn't seem to work as expected.

**long winded;**
Happy New Year everyone! This is my first post here and I'm a relative newbie to node so bare with me.

I've read all [the docs](https://nodejs.org/api/test.html) and even some seemingly related issues on the subject.

let's suppose I have tests (using `describe/it` pattern) as follows:

- a
  - length
- b
  - length
  
How do I match only `b>length`? I've tried every regex syntax I can find and there doesn't appear to be a way to do that. This forces me to uniquefy all test names which makes nested suites less useful than what they should be.

Thanks for all the hard work btw. Cool stuff.

**Comment by MoLow** [comment]
see https://github.com/nodejs/node/issues/47945

**Comment by rotty3000** [comment]
See my reply to that open issue https://github.com/nodejs/node/issues/47945#issuecomment-1877290974
